crypto/rsa/rsa_sign.c

```diff
@@ -91,6 +91,17 @@ int RSA_sign(int type, const unsigned char *m, unsigned int m_len,
         encoded_len = SSL_SIG_LENGTH;
         encoded = m;
     } else {
+        const EVP_MD *md = EVP_get_digestbynid(type);
+        if (md == NULL) {
+            RSAerr(RSA_F_RSA_SIGN, RSA_R_UNKNOWN_ALGORITHM_TYPE);
+            return 0;
+        }
```

Comment:
This is breaking compatibility. If the type does not resolve to an md, the m_len check should be simply skipped. Also please add empty line after declaration.

Reply:
Just pushed a commit to address this.